docs/getting-started/docker.rst

```diff
@@ -0,0 +1,190 @@
+available. (Reminder that this is still considered experimental, so apologies
+for this part being a bit strange.)
+
+You need to clone the `googleapis`_ repository from GitHub, and change to
+a special branch:
+
+.. code-block:: shell
+
+  $ git clone git@github.com:googleapis/googleapis.git
+  $ cd googleapis
+  $ git checkout --track -b input-contract origin/input-contract
+  $ cd ..
+
+The API available as an example (thus far) is the `Google Cloud Vision`_ API,
+available in the ``google/cloud/vision/v1/`` subdirectory. This will be used
+for the remainder of the examples on this page.
+
+.. _googleapis: https://github.com/googleapis/googleapis/tree/input-contract
+.. _Google Cloud Vision: https://cloud.google.com/vision/
+
+
+Compiling an API
+~~~~~~~~~~~~~~~~
+
+.. note::
+
+    If you are running code generation repeatedly, executing the
+    long ``docker run`` command may be cumbersome. While you should ensure
+    you understand this section, a :ref:`shortcut script<docker-shortcut>`
+    is available to make iterative work easier.
+
+Compile the API into a client library by invoking the Docker image.
+
+It is worth noting that the image must interact with the host machine
+(your local machine) for two things: reading in the protos you wish to compile,
+and writing the output. This means that when you run the image, two mount
+points are required in order for anything useful to happen.
+
+In particular, the input protos are expected to be mounted into ``/input/``,
+and the desired output location is expected to be mounted into ``/output/``.
+The output directory must also be writable.
+
+.. note::
+
+    The ``/input/`` and ``/output/`` directories inside the image are
+    hard-coded; they can not be altered where they appear in the command
+    below.
+
+Perform that step with ``docker run``:
+
+.. code-block:: shell
+
+    # This is assumed to be run from the `googleapis` project root.
+    $ docker run \
+      --mount type=bind,source=google/cloud/vision/v1/,destination=/input/google/cloud/vision/v1/,readonly \
+      --mount type=bind,source=dest/,destination=/output/ \
+      --rm \
+      --user $UID \
+      gcr.io/gapic-images/gapic-generator-python
+
+.. warning::
+
+    ``protoc`` is *very* picky about paths, and the exact construction here
+    matters a lot. The source is ``google/cloud/vision/v1/``, and then
+    the destination is that full directory path after the ``/input/`` root;
+    therefore: ``/input/google/cloud/vision/v1/``.
+
+    This matters because of how proto imports are resolved. The ``import``
+    statement imports a *file*, relative to a base directory or set of
+    base directories, called the ``proto_path``. This is assumed
+    (and hard-coded) to ``/input/`` in the Docker image, and so any directory
+    structure present in the imports of the proto files must be preserved
+    beneath this for compilation to succeed.
+
+
+Running a Client Library
```

Comment:
This the docker docs doesn't seem like the right place for this section. Maybe link out to it somewhere.

Reply:
This is just copied from `local.rst` outright. I agree with shifting it.

Reply:
Good call on this, thanks. I shifted the file and document structure around slightly and made it so that there is not (significant) repeated documentation. This will make it much easier to keep things up to date.

I do like the narrative flow of having verification on the getting started page, but it was subsumed under "usage" in a way that did not make sense. Fixed that.